### DISCUSSION TITLE
signInWithGoogle isn't working for me

### DISCUSSION THREAD

**Discussion by 34fame** [robingenz/capacitor-firebase-authentication#128] [Q&A] [answered]
When I call `await FirebaseAuthentication.signInWithGoogle()` I don't get a response.  No errors either.  In the console log in Xcode I see " To Native ->  FirebaseAuthentication signInWithGoogle 111223879" so the call is being submitted.  I'm running the app on my phone tethered to my laptop.

In my capacitor.config.json I have:
```
"plugins": {
      "FirebaseAuthentication": {
         "skipNativeAuth": false,
         "provider": ["google.com"]
      }
   }
```
My dependencies in the capacitor client are:
```
"dependencies": {
    "@capacitor-community/fcm": "^2.0.2",
    "@capacitor/cli": "^3.0.0",
    "@capacitor/core": "^3.0.0",
    "@capacitor/ios": "^3.0.0-beta.0",
    "@capacitor/local-notifications": "^1.1.0",
    "@capacitor/push-notifications": "^1.0.9",
    "@robingenz/capacitor-firebase-authentication": "^0.4.1",
    "firebase": "^9.6.5"
  }
```

I've followed all of the steps and double-checked everything.  I can't figure out what I'm missing or if maybe it's a version conflict.
![Screen Shot 2022-01-31 at 10 40 30 AM](https://user-images.githubusercontent.com/49797484/151835149-585dedac-ff18-4ad3-9eeb-9cbb072017f7.png)

Thanks!

**Comment by robingenz** [comment, marked as answer]
Hi @34fame,
thank you for your request! 
Please check the following:
1. [ ] `npm install @robingenz/capacitor-firebase-authentication firebase`
2. [ ] `npx cap sync`
3. [ ] Add Firebase to your project if you haven't already: https://firebase.google.com/docs/ios/setup
4. [ ] Verify that this function is included in your app's `AppDelegate.swift`: 
	  ```swift
	  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
	    return ApplicationDelegateProxy.shared.application(app, open: url, options: options)
	  }
	  ```
5. [ ] Add the `RobingenzCapacitorFirebaseAuthentication/Google` pod to your `Podfile` (usually `ios/App/Podfile`): 
	  ```diff
	  target 'App' do
	  capacitor_pods
	  # Add your Pods here
	  +  pod 'RobingenzCapacitorFirebaseAuthentication/Google', :path => '../../node_modules/@robingenz/capacitor-firebase-authentication'
	  end
	  ```
6. [ ] `npx cap update`
7. [ ] Add custom URL schemes to your Xcode project (see https://github.com/robingenz/capacitor-firebase-authentication/blob/main/docs/setup-google.md#ios)

If this is all met and still doesn't work, then please provide a Minimal, Reproducible Example (see [How to create a Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)) so i can debug it.

**Comment by santosbj** [comment]
Hi, i have the same problem!

I don't get a response after call await FirebaseAuthentication.signInWithGoogle() . No errors reported in console log.

I attached a test code.

[Archive.zip](https://github.com/robingenz/capacitor-firebase-authentication/files/8211950/Archive.zip)


Thanks for your help!

**Reply by robingenz** [reply]
Hi @santosbj, have you already checked what [I commented above](https://github.com/robingenz/capacitor-firebase-authentication/discussions/128#discussioncomment-2083843)?
If so, please upload your code to a GitHub repo and send me the link. It's easier that way.

**Comment by santosbj** [comment]
Hi @robingenz, 

the link: https://github.com/santosbj/robigenz-capacitor-test

Thanks!

**Reply by robingenz** [reply]
Please check my [comment above](https://github.com/robingenz/capacitor-firebase-authentication/discussions/128#discussioncomment-2083843).

Here's what's missing in your example (see [this line](https://github.com/santosbj/robigenz-capacitor-test/blob/54f9ebda503d74a889c46bc65726254b70e1f51e/ios/App/Podfile#L32) in your example):



Add the `RobingenzCapacitorFirebaseAuthentication/Google` pod to your `Podfile` (usually `ios/App/Podfile`): 
```diff
target 'App' do
capacitor_pods
# Add your Pods here
+  pod 'RobingenzCapacitorFirebaseAuthentication/Google', :path => '../../node_modules/@robingenz/capacitor-firebase-authentication'
end
```



Please fix this and try again.

**Reply by santosbj** [reply]
Hi @robingenz,

Yes, that was the problem. Huge mistake on me!

You're the best.

Thanks

**Comment by 34fame** [comment]
https://github.com/34fame/q-capacitor-sandbox

I'm using Quasar Framework (Vue.js) which works a little differently than a vanilla Capacitor project.  To run using Firebase emulators you need to run `yarn serve` in the `server/functions` directory and run `quasar dev -m capacitor -T ios -i` in the `client` directory.  Hopefully, the rest is familiar.

The login call is in `client/src/components/GoogleLoginBtn.vue`.  There, when I log `FirebaseAuthentication` I get an empty object.  Nothing happens when `await FirebaseAuthentication.signInWithGoogle()` is called.

**Reply by robingenz** [reply]
Same problem here.
Add the `RobingenzCapacitorFirebaseAuthentication/Google` pod to your `Podfile` (https://github.com/34fame/q-capacitor-sandbox/blob/main/client/src-capacitor/ios/App/Podfile): 
```diff
target 'App' do
capacitor_pods
# Add your Pods here
+  pod 'RobingenzCapacitorFirebaseAuthentication/Google', :path => '../../node_modules/@robingenz/capacitor-firebase-authentication'
end
```

It should look like that: 

```
platform :ios, '12.0'
use_frameworks!

# workaround to avoid Xcode caching of Pods that requires
# Product -> Clean Build Folder after new Cordova plugins installed
# Requires CocoaPods 1.6 or newer
install! 'cocoapods', :disable_input_output_paths => true

def capacitor_pods
  pod 'Capacitor', :path => '../../node_modules/@capacitor/ios'
  pod 'CapacitorCordova', :path => '../../node_modules/@capacitor/ios'
  pod 'CapacitorCommunityFcm', :path => '../../node_modules/@capacitor-community/fcm'
  pod 'CapacitorLocalNotifications', :path => '../../node_modules/@capacitor/local-notifications'
  pod 'CapacitorPushNotifications', :path => '../../node_modules/@capacitor/push-notifications'
  pod 'RobingenzCapacitorFirebaseAuthentication', :path => '../../node_modules/@robingenz/capacitor-firebase-authentication'
end

target 'App' do
  capacitor_pods
  # Add your Pods here
  pod 'RobingenzCapacitorFirebaseAuthentication/Google', :path => '../../node_modules/@robingenz/capacitor-firebase-authentication'
end
```

Please fix this and try again.

**Comment by 34fame** [comment]
Finally, after doing a diff I saw I was missing `/Google`!  Working now!
